lightning/src/util/events.rs

```diff
@@ -345,6 +345,22 @@ pub enum MessageSendEvent {
 	PaymentFailureNetworkUpdate {
 		/// The channel/node update which should be sent to NetGraphMsgHandler
 		update: msgs::HTLCFailChannelUpdate,
+	},
+	/// Used to send a query_channel_range message to the peer when a node wants to learn about
+	/// channels that exist in a given block range.
+	SendChannelRangeQuery {
+		/// The node_id of the node which should receive this message
+		node_id: PublicKey,
+		/// The message which should be sent.
+		msg: msgs::QueryChannelRange,
+	},
+	/// Used to send a query_short_channel_ids message to a peer to request routing gossip messages
+	/// for list of channels identified by their short_channel_ids.
+	SendChannelsQuery {
+		/// The node_id of the node which should receive this message
+		node_id: PublicKey,
+		/// The message which should be sent.
+		msg: msgs::QueryShortChannelIds,
```

Comment:
I think a worthy follow-up would be to coordinate better between query-gossips-at-connection and query-gossips-when-network-view-is-bad, AFAICT this patchset aims to query for each connection, which might be too much bandwidth hungry. I was thinking you could have a `timer_refresh_network_view_with_queries` in NetGraphMsgHandler on the model of PeerHandler::timer_tick_occured()

Reply:
Ah yes, sorry about that. I was doing some dev testing and left code that performs a sync on each connection. I still need to clean up the PeerHandler code to respect `should_request_full_sync`.  

Thanks, I'll check out `PeerHandler::timer_tick_occured()`.  For periodic anti-entropy, it might be nice to connect to a random node from the network view and attempt a sync.